scrcpy/core.py

```diff
@@ -239,6 +246,11 @@ def __stream_loop(self) -> None:
                 time.sleep(0.01)
                 if not self.block_frame:
                     self.__send_to_listeners(EVENT_FRAME, None)
+            except ConnectionAbortedError as e:
+                if self.alive:
+                    self.__send_to_listeners(EVENT_DISCONNECT)
+                    self.stop()
+                    raise e
             except OSError as e:  # Socket Closed
                 if self.alive:
                     self.stop()
```

Comment:
And send the event to the listener here since OSError also means the device is physically disconnected.